### PR TITLE
new package: sioyek

### PR DESCRIPTION
**Pull request by dkwo**
Sioyek is a PDF viewer designed for reading research papers and technical books. https://sioyek.info/
I applied a patch from a PR to use system mupdf, otherwise it would build its own.
I translated their build_linux.sh into our qmake build helper.
Added required package mujs.

### CODE REVIEW

**Comment by dkwo** [comment]
Not sure why it depends on gcc;
still not correct path in etc for some configs;
hack for hostmakedepends to see if cross work.

**Comment by abenson** [comment]
For each commit, please follow the [conventions](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes) for the commit message.

- `New package: mujs-1.2.0`
- `New package: sioyek-1.4.0`

File: srcpkgs/sioyek/template [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Template file for 'sioyek'
+pkgname=sioyek
+version=1.4.0
+revision=1
+build_style=qmake
+#make_build_args="pdf_viewer_build_config.pro"
+#make_install_args="sioyek pdf_viewer/prefs.config pdf_viewer/prefs_user.config
+# pdf_viewer/keys.config pdf_viewer/keys_user.config pdf_viewer/shaders
+# tutorial.pdf"
+hostmakedepends="zlib-devel qt5-qmake qt5-host-tools mupdf-devel
```
Comment (abenson) [inline srcpkgs/sioyek/template:10]: These `-devel` packages should be in `makedepends`.

Comment (dkwo) [reply srcpkgs/sioyek/template]: Done.

**Comment by dexgs** [comment]
Could mujs be moved into its own PR? I ask because its absence also blocks updating zathura-pdf-mupdf to 0.3.9.

**Comment by dkwo** [comment]
Sure, see #39167

File: srcpkgs/sioyek/template [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Template file for 'sioyek'
+pkgname=sioyek
+version=1.4.0
+revision=1
+build_style=qmake
+#make_build_args="pdf_viewer_build_config.pro"
```
Comment (abenson) [inline srcpkgs/sioyek/template:6]: Remove these comments.  Same with the `configure_args` further down.

**Comment by dkwo** [comment]
Done.
I'm still unable to cross compile though.

**Comment by leahneukirchen** [comment]
Sorry, you need to depend on `mujs-devel` now.

**Pull request merged**